### PR TITLE
Upgrade golangci-lint to v1.62.2

### PR DESCRIPTION
**Pull request by jeremybeard**
## Description

This change upgrades the golangci-lint dependency to the latest version so that it can support Go 1.23, which the CLI is also upgraded to. The corresponding linting errors are then also fixed.

## 📸 Screenshots

Screenshot 2024-12-12 at 12 37 01 PM

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: airflow/docker_image.go [hunk 1]
```diff
@@ -210,8 +210,8 @@ func (d *DockerImage) Pytest(pytestFile, airflowHome, envFile, testHomeDirectory
 
 	// start pytest container
 	err = cmdExec(containerRuntime, stdout, stderr, []string{"start", "astro-pytest", "-a"}...)
-	if docErr != nil {
-		log.Debugf("Error starting pytest container: %s", docErr.Error())
+	if err != nil {
```
Comment (neel-astro) [inline airflow/docker_image.go:213]: good catch 👍

File: airflow/runtimes/docker_runtime.go [hunk 2]
```diff
@@ -68,15 +68,15 @@ func (rt DockerRuntime) initializeDocker(timeoutSeconds int) error {
 	// If we got an error, Docker is not running, so we attempt to start it.
 	_, err = rt.Engine.Start()
 	if err != nil {
-		return fmt.Errorf(dockerOpenNotice) //nolint:stylecheck
+		return errors.New(dockerOpenNotice)
```
Comment (neel-astro) [inline airflow/runtimes/docker_runtime.go:71]: nit: we should stay with `fmt.Errorf` I think, since `pkg/errors` is deprecated now, wdyt?

**Review by neel-astro — APPROVED** [review]
LGTM, left a minor nit around `pkg/errors` vs `fmt.Errorf`, but approving ahead of time,

File: airflow/runtimes/docker_runtime.go (hunk 2 above)
Comment (jeremybeard) [reply airflow/runtimes/docker_runtime.go:71]: I can change this instance from `github.com/pkg/errors` to `errors`?

We might want a separate cleanup PR at some point to remove `github.com/pkg/errors` completely from the CLI, I see many instances of it, and in ways that can't be transparently switched to `errors`.

**Pull request merged**